app/api/items/enter.go

```diff
@@ -54,11 +58,25 @@ import (
 //   "500":
 //     "$ref": "#/responses/internalErrorResponse"
 func (srv *Service) enter(w http.ResponseWriter, r *http.Request) service.APIError {
-	parentAttemptID, err := service.ResolveURLQueryPathInt64Field(r, "attempt_id")
+	ids, err := idsFromRequest(r)
 	if err != nil {
 		return service.ErrInvalidRequest(err)
 	}
 
+	parentAttemptID, err := service.ResolveURLQueryGetInt64Field(r, "parent_attempt_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	user := srv.GetUser(r)
+	groupID := user.GroupID
+	if len(r.URL.Query()["as_team_id"]) != 0 {
```

Comment:
I am not sure you validate that the user is allowed to use this team.  As written in #496 , I wonder if it wouldn't be cleaner (less repetition, handled once for all) and safer (make sure we never forget to check) to handle it in a middleware (enabled for some services only) which would check the perm (at the risk of doing sometimes one extrarequest where one would be enough).
It would also to just check have 1 test verifying the group is used (the happy path), without requiring the error case in every service.

Reply:
getItemInfoAndEntryState() checks this (via GroupStore.TeamGroupForUser()).
Sure, it will be done in a middleware (#496 is on my list).

Reply:
Ok, feel free to start #496 at any convenient moment.